ReactWindows/ReactNative.Shared/UIManager/ViewManager.cs

```diff
@@ -107,6 +108,12 @@ public void UpdateProperties(TFrameworkElement viewToUpdate, ReactStylesDiffMap
         public TFrameworkElement CreateView(ThemedReactContext reactContext)
         {
             var view = CreateViewInstance(reactContext);
+#if WINDOWS_UWP
+            if (view is UIElement uiElement)
+            {
+                AccessibilityHelper.OnViewInstanceCreated(uiElement);
+            }
```

Comment:
Do we want to set a name on *every* view created in react-native, or should we be more selective? XAML doesn't create a name for every element...

Reply:
@rozele  The reason for this is that some UIElements e.g. Border, Canvas, etc. do not create automation peer immediately, but only if the name was set at least once (then the peer is cached for the lifetime of the element). For our implementation (since it relies on traversing the automation tree) we must guarantee that automation peer is always created. To do this we set the automationname to space and then clear it. This is not going to affect the element that already create automation peer e.g. Button, but will guarantee that all the elements will always create a peer. The label is cleared to be empty that is the default value 

---
In reply to: [170249495](https://github.com/Microsoft/react-native-windows/pull/1645#discussion_r170249495) [](ancestors = 170249495)

Reply:
Is there a significant performance hit from this?

---
In reply to: [170340069](https://github.com/Microsoft/react-native-windows/pull/1645#discussion_r170340069) [](ancestors = 170340069,170249495)

Reply:
@rozele We considered performance. Since the element already exists in the visual tree and the automation tree is just a subset of it and also this will be the case only for some of the elements, we thought that the performance hit will be minimal - it will just create some extra automationpeers in some cases that may take some extra memory

---
In reply to: [170552436](https://github.com/Microsoft/react-native-windows/pull/1645#discussion_r170552436) [](ancestors = 170552436,170340069,170249495)